hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2799,6 +2800,33 @@ class Emit[C](
               }
             })
           }
+        case NDArrayAgg(child, axesToSumOut) =>
+          deforest(child).map(cb) { childEmitter =>
+            val childDims = child.typ.asInstanceOf[TNDArray].nDims
+            val axesToKeep = (0 until childDims).filter(axis => !axesToSumOut.contains(axis))
+            val newOutputShape = axesToKeep.map(idx => childEmitter.outputShape(idx))
+            val newOutputShapeComplement = axesToSumOut.map(idx => childEmitter.outputShape(idx))
+
+            new NDArrayEmitter(newOutputShape) {
+              override def outputElement(cb: EmitCodeBuilder, idxVars: IndexedSeq[Value[Long]]): PCode = {
+                val numericElementType = coerce[PNumeric](child.pType.asInstanceOf[PNDArray].elementType)
+                val runningSum = coerce[Any](cb.newLocal("running_sum")(typeToTypeInfo(numericElementType)))
+                cb.assign(runningSum, numericElementType.zero)
```

Comment:
@tpoterba, is this hacky type info stuff still the right way to handle situations like this?